test/e2e/shoot/create_and_delete_unprivileged.go

```diff
@@ -16,30 +16,101 @@ package shoot
 
 import (
 	"context"
+	"os"
 	"time"
 
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	. "github.com/gardener/gardener/pkg/utils/test/matchers"
 	"github.com/gardener/gardener/test/e2e"
+	"github.com/gardener/gardener/test/framework"
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/utils/pointer"
 )
 
 var _ = Describe("Shoot Tests", Label("Shoot", "default"), func() {
-	f := defaultShootCreationFramework()
-	f.Shoot = e2e.DefaultShoot("unpriv-")
+	f := framework.NewShootCreationFramework(&framework.ShootCreationConfig{
+		GardenerConfig: &framework.GardenerConfig{
+			ProjectNamespace:   projectNamespace,
+			GardenerKubeconfig: os.Getenv("KUBECONFIG"),
+			SkipAccessingShoot: false,
+		},
+	})
+
+	f.Shoot = e2e.DefaultShoot("")
+	f.Shoot.Name = "e2e-unprivileged"
 	f.Shoot.Spec.Kubernetes.AllowPrivilegedContainers = pointer.Bool(false)
+	f.Shoot.Spec.Kubernetes.KubeAPIServer = &gardencorev1beta1.KubeAPIServerConfig{
+		AdmissionPlugins: []gardencorev1beta1.AdmissionPlugin{
+			{
+				Name:   "PodSecurity",
+				Config: getConfig(),
+			},
+		},
+	}
 
 	It("Create and Delete Unprivileged Shoot", Label("unprivileged"), func() {
 		By("Create Shoot")
 		ctx, cancel := context.WithTimeout(parentCtx, 15*time.Minute)
 		defer cancel()
+
 		Expect(f.CreateShootAndWaitForCreation(ctx, false)).To(Succeed())
 		f.Verify()
 
+		shootClient := f.ShootFramework.ShootClient.Client()
+
+		By("Create pod in the kube-system namespace")
+		podInKubeSystem := getPod(metav1.NamespaceSystem)
+		Expect(shootClient.Create(ctx, podInKubeSystem)).To(Succeed())
+
+		By("Create pod in the default namespace")
+		podInDefault := getPod(metav1.NamespaceDefault)
+		err := shootClient.Create(ctx, podInDefault)
+		Expect(err).To(BeForbiddenError())
+		Expect(err).To(MatchError(ContainSubstring("pods %q is forbidden: violates PodSecurity %q", podInDefault.Name, "restricted:latest")))
+
 		By("Delete Shoot")
 		ctx, cancel = context.WithTimeout(parentCtx, 15*time.Minute)
 		defer cancel()
 		Expect(f.DeleteShootAndWaitForDeletion(ctx, f.Shoot)).To(Succeed())
 	})
 })
+
+func getConfig() *runtime.RawExtension {
+	admissionConfigData := []byte(`{
+	  "apiVersion": "pod-security.admission.config.k8s.io/v1beta1",
+	  "kind": "PodSecurityConfiguration",
+	  "defaults": {
+	    "enforce": "restricted",
+	    "enforce-version": "latest"
+	  }
+    }`)
+
+	return &runtime.RawExtension{Raw: admissionConfigData}
+}
+
+func getPod(namespace string) *corev1.Pod {
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "nginx",
+			Namespace: namespace,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:  "nginx",
+					Image: "nginx:1.14.2",
+					Ports: []corev1.ContainerPort{
+						{
+							ContainerPort: 80,
+						},
+					},
+				},
+			},
+		},
+	}
```

Comment:
Why is this pod spec causing a forbidden error? What is the problematic part?

Reply:
When `restricted` profile is set, 
```
Error from server (Forbidden): error when creating "sample-pod.yaml": pods "nginx" is forbidden: violates PodSecurity "restricted:latest": allowPrivilegeEscalation != false (container "nginx" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "nginx" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "nginx" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "nginx" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```

Reply:
Oh wow, this is indeed quite restrictive. I am asking myself whether we should default `allowPrivilegedContainers=false` for `1.25+` shoots (#6570). It rather feels as if this should be a conscious decision by the cluster owner?

Reply:
We could also do `baseline` level, which gives the minimal privileges.
But the motive behind this particular e2e test is, even if the user sets a config (`restricted` or whatever) for the new plugin, we should be able to deploy our pods.